wp-content/civi-extensions/goonjcustom/Civi/CollectionBaseService.php

```diff
@@ -62,6 +64,20 @@ public static function checkIfPosterNeedsToBeGenerated($op, $objectName, $id, &$
 
   }
 
+  /**
+   *
+   */
+  private static function generateBaseFileName($collectionSourceId) {
```

Comment:
Two things:
1. If you create a function like this it should be used for both collection camp service and dropping center service
2. This file `CollectionBaseService` should not have anything specific to collection camp. It should be the responsibility of the individual classes: `CollectionCampService` and `DroppingCenterService` to define what should be the file name.

Reply:
We can use a switch-case statement to handle different entity subtypes in the base class, like this:

```
public static function generateBaseFileName($collectionSourceId) {
    $entitySubtypeName = self::getEntitySubtypeName($collectionSourceId);
    error_log("entitySubtypeName: " . print_r($entitySubtypeName, TRUE));

    switch ($entitySubtypeName) {
        case 'Dropping_Center':
            return DroppingCenterService::generatePosterFileName($collectionSourceId);
        case 'Collection_Camp':
            return CollectionCampService::generatePosterFileName($collectionSourceId);
        default:
            throw new Exception("Unknown entity subtype: " . $entitySubtypeName);
    }
}
```
in the Dropping Center or Collection Camp services, we can define methods to handle file naming:

```
public static function generatePosterFileName($collectionSourceId) {
    return "dropping_center_{$collectionSourceId}.png";
}
```